tests/test_suite.py

```diff
@@ -226,76 +214,37 @@ def monitorOutputFile(finalMessage):
     monitorOutputCommand.run(expectedString=finalMessage, silent=False)
 
 
-if not args.nomake:
-    executeMake()
-setupWebots()
-resetOutputFile()
-
-finalMessage = 'Test suite complete'
-thread = threading.Thread(target=monitorOutputFile, args=[finalMessage])
-thread.start()
-
-webotsArguments = '--mode=fast --stdout --stderr --batch'
-webotsArgumentsNoRendering = webotsArguments + ' --no-rendering --minimize'
-
-
-for groupName in testGroups:
-    if groupName == 'cache':
-        update_cache_urls()  # setup new environment
-
-    testFailed = False
-
-    appendToOutputFile('\n### ' + groupName + ' test\n\n')
+def runGroupTest(groupName, firstSimulation, worldsCount, failures):
+    if not os.path.exists(firstSimulation):
+        return
 
     # clear stdout and stderr files
     open(webotsStdErrFilename, 'w').close()
     open(webotsStdOutFilename, 'w').close()
 
-    worldsFilename = testsFolderPath + groupName + os.sep + 'worlds.txt'
-    indexFilename = testsFolderPath + groupName + os.sep + 'worlds_index.txt'
-
     # init temporary world counter file
     tempFile = open(tempWorldCounterFilename, 'w')
     tempFile.write('0')
     tempFile.close()
 
-    supervisorTargetDirectory = testsFolderPath + groupName + os.sep + 'controllers' + os.sep + \
-        supervisorControllerName
-    if not os.path.exists(supervisorTargetDirectory):
-        os.makedirs(supervisorTargetDirectory)
-    shutil.copyfile(
-        defaultProjectPath + 'controllers' + os.sep +
-        supervisorControllerName + os.sep +
-        supervisorControllerName + '.py',
-        supervisorTargetDirectory + os.sep + supervisorControllerName + '.py'
-    )
-    # parser tests uses a slightly different Supervisor PROTO
-    protosTargetDirectory = testsFolderPath + groupName + os.sep + 'protos'
-    protosSourceDirectory = defaultProjectPath + 'protos' + os.sep
-    if not os.path.exists(protosTargetDirectory):
-        os.makedirs(protosTargetDirectory)
-    for protoFileName in protoFileNames:
-        shutil.copyfile(protosSourceDirectory + protoFileName,
-                        protosTargetDirectory + os.sep + protoFileName)
-    worldsCount = generateWorldsList(groupName, worldsFilename)
-    firstSimulation = findFirstWorldFilename(worldsFilename)
-    if not os.path.exists(firstSimulation):
-        continue
-
+    indexFilename = testsFolderPath + groupName + os.sep + 'worlds_index.txt'
```

Comment:
Why not using [os.path.join](https://docs.python.org/3/library/os.path.html#os.path.join) which [seems to be a better practice](https://stackoverflow.com/a/16789790/810268) than os.sep? (this could apply to other places in the code)
```suggestion
    indexFilename = os.path.join(testsFolderPath, groupName, 'worlds_index.txt')
```